`mypy -p oaff` without errors supressed:
<details>
<summary>mypy output</summary>

```
oaff) ysebastien@oscar:~/git/ogc-api-fast-features$ mypy -p oaff
oaff/app/oaff/app/request_handlers/landing_page.py:15: error: Name "LandingPage" already defined (possibly by an import)
oaff/app/oaff/app/request_handlers/landing_page.py:20: error: Missing return statement
oaff/app/oaff/app/request_handlers/conformance.py:10: error: Name "Conformance" already defined (possibly by an import)
oaff/app/oaff/app/request_handlers/conformance.py:15: error: Missing return statement
oaff/app/oaff/app/data/sources/common/data_source_manager.py:9: error: Method must have at least one argument
oaff/app/oaff/app/responses/response_format.py:6: error: Definition of "__hash__" in base class "dict" is incompatible with definition in base class "Enum"
oaff/app/oaff/app/request_handlers/feature.py:22: error: Missing return statement
oaff/app/oaff/app/request_handlers/collections_list.py:15: error: Name "CollectionsList" already defined (possibly by an import)
oaff/app/oaff/app/request_handlers/collections_list.py:20: error: Missing return statement
oaff/app/oaff/app/request_handlers/collection_items.py:36: error: Missing return statement
oaff/app/oaff/app/request_handlers/collection_items.py:102: error: Incompatible return value type (got "Dict[Any, Any]", expected "List[Any]")
oaff/app/oaff/app/request_handlers/collection_items.py:104: error: Incompatible return value type (got "Callable[[int, int], List[Any]]", expected "Callable[[int, int], Dict[Any, Any]]")
oaff/app/oaff/app/request_handlers/collection_items.py:112: error: Incompatible return value type (got "None", expected "Type[Any]")
oaff/app/oaff/app/request_handlers/collection_items.py:134: error: Too many values to unpack (4 expected, 6 provided)
oaff/app/oaff/app/request_handlers/collection_items.py:134: error: "None" object is not iterable
oaff/app/oaff/app/request_handlers/collection_items.py:135: error: Argument 1 to "len" has incompatible type "Union[Tuple[float, float, float, float], Tuple[float, float, float, float, float, float], None]"; expected "Sized"
oaff/app/oaff/app/request_handlers/collection_items.py:136: error: Value of type "Union[Tuple[float, float, float, float], Tuple[float, float, float, float, float, float], None]" is not indexable
oaff/app/oaff/app/request_handlers/collection_items.py:160: error: Incompatible return value type (got "None", expected "Type[Any]")
oaff/app/oaff/app/request_handlers/collection_items.py:164: error: Need more than 1 value to unpack (2 expected)
oaff/app/oaff/app/request_handlers/collection_items.py:312: error: Incompatible return value type (got "None", expected "Type[Any]")
oaff/app/oaff/app/request_handlers/collection_items.py:338: error: Argument 1 to "astimezone" of "datetime" has incompatible type "Type[tzinfo]"; expected "Optional[tzinfo]"
oaff/app/oaff/app/request_handlers/collection.py:19: error: Missing return statement
oaff/app/oaff/app/responses/templates/templates.py:16: error: "Environment" has no attribute "install_gettext_translations"
oaff/app/oaff/app/data/sources/postgresql/stac_hybrid/postgresql_data_source.py:74: error: "Coroutine[None, None, Database]" not callable
oaff/app/oaff/app/data/sources/postgresql/stac_hybrid/postgresql_data_source.py:183: error: Item "None" of "Optional[Any]" has no attribute "limit"
oaff/app/oaff/app/data/sources/postgresql/stac_hybrid/postgresql_data_source.py:184: error: Item "None" of "Optional[Any]" has no attribute "offset"
oaff/app/oaff/app/data/sources/postgresql/stac_hybrid/postgresql_data_source.py:188: error: Value of type "Optional[Mapping[Any, Any]]" is not indexable
oaff/app/oaff/app/data/sources/postgresql/stac_hybrid/postgresql_data_source.py:226: error: List item 0 has incompatible type "List[None]"; expected "datetime"
oaff/app/oaff/app/data/sources/postgresql/stac_hybrid/postgresql_data_source.py:312: error: Missing return statement
oaff/app/oaff/app/data/sources/postgresql/stac_hybrid/postgresql_data_source.py:418: error: Need type annotation for "table_temporal_extents" (hint: "table_temporal_extents: Dict[<type>, <type>] = ...")
oaff/app/oaff/app/data/sources/postgresql/stac_hybrid/postgresql_data_source.py:441: error: Value of type "Optional[Mapping[Any, Any]]" is not indexable
oaff/fastapi/api/util.py:60: error: Dict entry 0 has incompatible type "str": "int"; expected "str": "str"
oaff/fastapi/api/util.py:61: error: Dict entry 1 has incompatible type "str": "int"; expected "str": "str"
oaff/fastapi/api/routes/common/common_parameters.py:87: error: Incompatible return value type (got "List[str]", expected "Set[str]")
oaff/fastapi/api/routes/common/common_parameters.py:87: error: Argument "key" to "sorted" has incompatible type overloaded function; expected "Callable[[str], SupportsLessThan]"
oaff/fastapi/api/openapi/openapi.py:41: error: Incompatible return value type (got "Callable[[Request], Any]", expected "VndResponse")
oaff/fastapi/api/routes/common/parameter_control.py:12: error: Unsupported left operand type for + ("None")
oaff/fastapi/api/routes/common/parameter_control.py:12: note: Left operand is of type "Optional[List[str]]"
oaff/fastapi/api/routes/collections.py:83: error: Incompatible default for argument "limit_param" (default has type "Query", argument has type "int")
oaff/fastapi/api/routes/collections.py:89: error: Incompatible default for argument "offset_param" (default has type "Query", argument has type "int")
oaff/fastapi/api/routes/collections.py:94: error: Incompatible default for argument "bbox_param" (default has type "Query", argument has type "Optional[str]")
oaff/fastapi/api/routes/collections.py:110: error: Incompatible default for argument "bbox_crs_param" (default has type "Query", argument has type "Optional[str]")
oaff/fastapi/api/routes/collections.py:114: error: Incompatible default for argument "datetime_param" (default has type "Query", argument has type "Optional[str]")
oaff/fastapi/api/routes/collections.py:126: error: Incompatible default for argument "filter_param" (default has type "Query", argument has type "Optional[str]")
oaff/fastapi/api/routes/collections.py:130: error: Incompatible default for argument "filter_lang_param" (default has type "Query", argument has type "Optional[str]")
oaff/fastapi/api/routes/collections.py:134: error: Incompatible default for argument "filter_crs_param" (default has type "Query", argument has type "Optional[str]")
oaff/fastapi/api/routes/collections.py:205: error: Incompatible return value type (got "None", expected "datetime")
oaff/fastapi/api/routes/collections.py:219: error: Incompatible return value type (got "Tuple[datetime, ...]", expected "Union[Tuple[datetime], Tuple[datetime, datetime], None]")
Found 47 errors in 15 files (checked 149 source files)
```

</details>

--- a/README.md
+++ b/README.md
@@ -115,7 +115,7 @@ The following provides information for developers looking to maintain or extend 
 ## Architecture
 
 ## Pre-commit hooks
-Git hooks are used to run formatting (isort, black), linting (flake8), and type-checking (mypy). You can install the pre-commit scripts with `pre-commit install`. Thereafter, the scripts defined in `.pre-commit-config.yaml` will be run on the files included in a commit. Specific tool configurations are included in `pyproject.toml`.
+Git hooks are used to run formatting (isort, black), linting (flake8), and type-checking (mypy). The pre-commit scripts can be installed by running `scripts/setup` or simply `pre-commit install`. Thereafter, the scripts defined in `.pre-commit-config.yaml` will be run on the files included in a commit. Specific tool configurations are included in `pyproject.toml`.
 
 ### FastAPI
 oaff's frontend uses FastAPI but the backend is not tightly-coupled to that framework. Frontend routers interpret path and query parameters and build request objects using request classes defined in the backend. A single `delegate` method passes request objects to the backend and awaits a response. Responses provide either data or error detail and an appropriate API response is constructed from whichever is returned. During the frontend's startup sequence a `FrontendConfiguration` object is passed to the backend, providing key frontend properties that need to be known by the backend. Examples:

--- a/mypy.sh
+++ b/mypy.sh
@@ -1,3 +1,0 @@
-#!/bin/sh
-
-mypy --namespace-packages -p oaff

